ivy/functional/frontends/torch/nn/functional/non_linear_activation_functions.py

```diff
@@ -356,6 +356,26 @@ def softplus(input, beta=1, threshold=20):
     return ivy.softplus(input, beta=beta, threshold=threshold)
 
 
+@to_ivy_arrays_and_back
+@with_unsupported_dtypes({"2.0.1 and below": ("float16",)}, "torch")
+def gumbel_softmax(logits, tau=1, hard=False, eps=1e-10, dim=-1):
+    gumbels = -ivy.empty_like(logits).exponential().log()
+    gumbels = (logits + gumbels) / tau
+    y_soft = ivy.softmax(x=gumbels, axis=dim)
+
+    if hard:
+        indices = y_soft.max(axis=dim, keepdims=True)[1]
+        y_hard = ivy.zeros_like(logits).scatter(dim, indices, 1.0)
+        updates = ivy.ones_like(indices)
+        ivy.scatter_nd(indices, updates, reduction="replace", out=y_hard)
```

Comment:
Any reason as to why the scatter_nd needed here?

Reply:
.scatter(dim, indices, 1.0) was redundant as there is no ivy implementation of that function. I had to either use scatter_flat or scatter_nd but since multidimensionality is required, I chose scatter_nd.